### ISSUE TITLE
[Plugin Request] Make Markdown suport as a plugin.

### ISSUE THREAD

**Issue by bccrisan** [mozilla-releng/firefox-infra-changelog#414]
We have decided to stop supporting markdown as the primary way of displaying the data, as right now everything runs based on the json (which has "raw" data in it).

The data will continue to be displayed as it is, the only difference it's that all markdown logic will be moved into a separate package (called plugin).

One of the reasons we want to do this is de-cluttering the main code and give an easy way of disable it, should you not need it.

**Comment by Akhliskun** [comment]
@bccrisan I have changed a few things in the description above as it sounded weird :)

**Comment by bccrisan** [comment]
Thanks for the update!

**Comment by raduiman** [comment]
I started working on this issue in the branch "issue-414"

**Comment by zsoltfay** [comment]
Continued working on this issue. Latest updates have been pushed to its branch.

Added 2 new methods. The first one creates the MD header and the first table row. The second one filters out, from the local json data, which commits we need for the MD file. See both below:

```python
def _build_initial_md_structure(self):
      self.md_ready_data.append(self._create_repo_markdown_header())
      self.md_ready_data.append(self._create_first_table_row())
```

```python
def _populate_md_table(self):
    local_json_data = self._load_local_json_data()
    if len(local_json_data) > 0:
         del local_json_data["0"]
         self.commit_number = 1
         for key in local_json_data:
             self.commit_date = local_json_data.get(key).get("date")
             if self.commit_date > return_time("%Y-%m-%dT%H:%M:%S", "sub", 7):
                  self.commit_author = local_json_data.get(key).get("author")
                  self.commit_url = local_json_data.get(key).get("url")
                  self.commit_message = local_json_data.get(key).get("message")
                  self.commit_author, self.commit_message = self.filter_strings()
                  self.trim_commit_description(self.commit_url, COMMIT_DESCRIPTION_LENGTH)
                  self.generate_link_for_bugs()
                  self.md_ready_data.append(self.md_table_row_builder())
                  self.commit_number += 1
```
Also created an entry point:
```python
def start_md_for_git(self, repo_name=None):
    self.repo_name = repo_name
    self._build_initial_md_structure()
    self._populate_md_table()
    for element in a.md_ready_data:
        print(element)
        a.save(CHANGELOG_REPO_PATH, a.repo_name + ".md", element)
```
If you wish to test these changes make sure you have the json files present locally and run:
```python
a = FICMarkdownGenerator()
a.start_md_for_git("taskcluster")
```

**Comment by zsoltfay** [comment]
In the attempt to write the logic that creates changelog.md I implemented the code that populates changelod.json. You can find it in FIC_Core.py
 =======================================================
It is expecting one argument: the number of days that we want to go back when we will be listing the commits in changelog.md
 
There are 2 methods `_extract_git_commits` and `_extract_hg_commits` that populate a dict called `changelog`, which is then written into changelog.json

**Comment by raduiman** [comment]
It was implemented in PR #465

**Comment by raduiman** [comment]
I'll reopen this issue since there are a few more things to be implemented.
- Issue #466 
- incorporate the logic in FIC_Core.py  to auto generate the MD files along with .json files
- add the logging system
